pkg/api/datasources.go

```diff
@@ -351,28 +352,57 @@ func validateJSONData(ctx context.Context, jsonData *simplejson.Json, cfg *setti
 
 	// Prevent adding a data source team header with a name that matches the auth proxy header name
 	if features.IsEnabled(featuremgmt.FlagTeamHttpHeaders) {
-		teamHTTPHeadersJSON, err := datasources.GetTeamHTTPHeaders(jsonData)
+		err := validateTeamHTTPHeaderJSON(jsonData)
 		if err != nil {
-			datasourcesLogger.Error("Unable to marshal TeamHTTPHeaders")
-			return errors.New("validation error, invalid format of TeamHTTPHeaders")
+			return err
 		}
-		// whitelisting X-Prom-Label-Policy
-		for _, headers := range teamHTTPHeadersJSON {
-			for _, header := range headers {
-				// TODO: currently we only allow for X-Prom-Label-Policy header to be used by our proxy
-				for _, name := range []string{"X-Prom-Label-Policy"} {
-					if http.CanonicalHeaderKey(header.Header) != http.CanonicalHeaderKey(name) {
-						datasourcesLogger.Error("Cannot add a data source team header that is different than", "headerName", name)
-						return errors.New("validation error, invalid header name specified")
-					}
+	}
+
+	return nil
+}
+
+func validateTeamHTTPHeaderJSON(jsonData *simplejson.Json) error {
+	teamHTTPHeadersJSON, err := datasources.GetTeamHTTPHeaders(jsonData)
+	if err != nil {
+		datasourcesLogger.Error("Unable to marshal TeamHTTPHeaders")
+		return errors.New("validation error, invalid format of TeamHTTPHeaders")
+	}
+	// whitelisting ValidHeaders
+	for _, headers := range teamHTTPHeadersJSON {
+		for _, header := range headers {
+			for _, name := range ValidHeaders() {
+				if http.CanonicalHeaderKey(header.Header) != http.CanonicalHeaderKey(name) {
+					datasourcesLogger.Error("Cannot add a data source team header that is different than", "headerName", name)
+					return errors.New("validation error, invalid header name specified")
+				}
+				if !TeamHTTPHeaderValueRegexMatch(header.Value) {
+					datasourcesLogger.Error("Cannot add a data source team header value with invalid value", "headerValue", header.Value)
+					return errors.New("validation error, invalid header value syntax")
 				}
 			}
 		}
 	}
-
 	return nil
 }
 
+// TeamHTTPHeaderValueRegexMatch returns a regex that can be used to check
+// words separated by special characters
+// "namespace!=auth", "env="prod", "env!~dev"
+func TeamHTTPHeaderValueRegexMatch(headervalue string) bool {
```

Comment:
We can make it non-exported since it's not used anywhere else, to keep module more clean.
```suggestion
func teamHTTPHeaderValueRegexMatch(headervalue string) bool {
```